packages/pyright-internal/src/tests/samples/based_self_default.py

```diff
@@ -0,0 +1,24 @@
+# pyright: reportUnusedParameter=none
+
+# The `self` or `cls` parameter of a method should not have a default value
+
+
+class Foo:
+    def instance_method(self=42) -> None:
+        pass
+
+    @classmethod
+    def class_method(cls=42) -> None:
+        pass
+
+    @staticmethod
+    def static_method(something=42) -> None:
```

Comment:
can we also have a test to ensure there's no error when the first argument is keyword-only?

```py
    @staticmethod
        def static_method(*, something=42):
            ...
```